R/create_individual_file.R

```diff
@@ -760,3 +762,28 @@ clean_up_gender <- function(individual_file) {
       )
     )
 }
+
+#' Join slf lookup variables
+#'
+#' @description Join lookup variables from slf postcode lookup and slf gpprac
+#'              lookup.
+#'
+#' @param individual_file the processed individual file.
+join_slf_lookup_vars <- function(individual_file) {
+  individual_file <- individual_file %>%
+    dplyr::left_join(
+      read_file(
+        get_slf_postcode_path()
+      ),
+      by = "postcode"
+    ) %>%
+    dplyr::left_join(
+      read_file(
+        get_slf_gpprac_path()
+      ) %>%
+        dplyr::select(c("gpprac", "cluster", "hbpraccode")),
+      by = "gpprac"
+    )
+
+  return(individual_file)
+}
```

Comment:
So it works nicely with targets, could we have the lookups passed as parameters please e.g.

```suggestion
join_slf_lookup_vars <- function(
  individual_file,
  slf_postcode_lookup = read_file(get_slf_postcode_path()),
  slf_gpprac_lookup = read_file(
                       get_slf_gpprac_path(), 
                       col_select = c("gpprac", "cluster", "hbpraccode")
                       )
  ) {
  individual_file <- individual_file %>%
    dplyr::left_join(
      slf_postcode_lookup,
      by = "postcode"
    ) %>%
    dplyr::left_join(
      slf_gpprac_lookup %>%
        dplyr::select("gpprac", "cluster", "hbpraccode"),
      by = "gpprac"
    )

  return(individual_file)
}
```

Reply:
I think we can resolve the `hbrescode` here by adding `dplyr::rename(hbrescode = "HB2018")` - in fact could we have the hbrescode variable as a parameter so it's simpler to change in the future? e.g. `dplyr::rename(hbrescode = hbrescode_var)`.